src/components/customer/appsco-add-partner-admin-to-customers.html

```diff
@@ -0,0 +1,600 @@
+            _onDialogOpened: function() {
+//                this.$.name.focus();
+                this._loadRoles();
+                Polymer.Gestures.add(this, 'tap', this._handleDocumentClick.bind(this));
+            },
+
+            /**
+             * Called after dialog has been closed.
+             *
+             * @private
+             */
+            _onDialogClosed: function() {
+                this._hideLoader();
+                this._hideError();
+                this._target.reset();
+                this.set('_administrators', []);
+                this.set('_selectedAdministrator', {});
+                this._hideAdminList();
+                this.$.appscoSearchAdmins.reset();
+                Polymer.Gestures.remove(this, 'tap', this._handleDocumentClick.bind(this));
+            },
+
+            _onAddAction: function() {
+                this._submitForm();
+            },
+
+            /**
+             * Submits form on ENTER key using iron-a11y-keys component.
+             *
+             * @private
+             */
+            _onEnterAction: function() {
+                this._onAddAction();
+            },
+
+            _submitForm: function() {
+                this._hideError();
+
+                if (!this._selectedAdministrator.value) {
+                    this._showError('You have not selected partner administrator.');
+                    this.$.appscoSearchAdmins.focusAppscoSearchInput();
+                    return false;
+                }
+
+                if (this._target.validate()) {
+                    this._showLoader();
+                    this._target.submit();
+                }
+
+            },
+
+            // TODO It isn't done - Customers should be added to body too
+            _onFormPresubmit: function(event) {
+                event.target.request.body['company_customer[partnerAdmin]'] = this._selectedAdministrator.value;
+            },
+
+            /**
+             * Called on form error.
+             *
+             * @param {Object} event
+             * @private
+             */
+            _onFormError: function(event) {
+                this._showError(this.apiErrors.getError(event.detail.request.response.code));
+                this._hideLoader();
+            },
+
+            /**
+             * Called after group has been added.
+             *
+             * @param {Object} event
+             * @private
+             */
+            _onFormResponse: function(event) {
+                this.close();
+
+                this.fire('partner-admin-added', {
+                    customer: event.detail.response
+                });
+
+            },
+
+            _onRolesError: function() {
+                this.set('_administrators', []);
+            },
+
+            _onRolesResponse: function(event) {
+                var response = event.detail.response;
+
+                if (!response || (response && response.meta && 0 === response.meta.total)) {
```

Comment:
No need for all of the code.

_onRolesResponse: function(event) {
                var response = event.detail.response;

                if (!response || (response && response.meta && 0 === response.meta.total)) {
                    this.set('_administrators', []);
                }
                else if (response && response.company_roles) {
                    this.set('_administrators', response.company_roles);
                }

                this.set('_administratorsToDisplay', this._administrators);
            },